WooCommerce/Classes/POS/ViewModels/PointOfSaleDashboardViewModel.swift

```diff
@@ -31,6 +31,8 @@ final class PointOfSaleDashboardViewModel: ObservableObject {
     /// This boolean is used to determine if the whole totals/payments view is occupying the full screen (cart is not showed)
     @Published var isTotalsViewFullScreen: Bool = false
     @Published var isInitialLoading: Bool = false
+    @Published var isError: Bool = false
+    @Published var isEmpty: Bool = false
```

Comment:
I think we can simplify this bit by either moving the isInitialLoading state to `ItemListState`, and then checking, or creating a new DashboardState with several cases (isInitialLoading, isLoaded, isError, isEmpty). By having these booleans we need to be sure to switch them accordingly when observing state, otherwise we could land on unexpected/wrong state. 

It would also simplify loading the views, since we can use a switch, and the `observeItemListState` method.

Thoughts? @staskus

Reply:
Yes, I think we can remove `ItemListState` for the view model, and have a similar functionality for the dashboard. `ItemList` no longer has error or loading states, we're showing everything in `PointOfSaleDashboardView`.

Reply:
Great! I started to refactor it but as shared I think we want to do this separately so we're sure to test it properly on its own PR. I've opened a follow-up here: https://github.com/woocommerce/woocommerce-ios/issues/13543